File: code/cryptography/src/vigenere_cipher/vigenere_cipher.php
<?php
/**
* Vigenère cipher (PHP)
* based on https://en.wikipedia.org/wiki/Vigen%C3%A8re_cipher
* implemented by Stefano Amorelli <amoste@outlook.it>
* Part of Cosmos by OpenGenus Foundation
*/

class VigenereCipher
{
 public function encrypt($plainText, $key) {

  if (!ctype_alpha($plainText) || !ctype_alpha($key)) {
   throw new Exception('Plaintext and key should be alphabetical strings (no spaces, symbols and numbers).');
  }

  $plainText  = strtolower($plainText);
  $key        = strtolower($key);
  $encryptedText = [];
  $i          = 0;

  foreach (str_split($plainText) as $character) {
   array_push($encryptedText, $this->encryptCharacter($character, $key[$i]));
   $i++;
   $i = $i % strlen($key);
  }
  return implode('', $encryptedText);

 }

 public function decrypt($encryptedText, $key) {

  if (!ctype_alpha($encryptedText) || !ctype_alpha($key)) {
   throw new Exception('Encrypted text and key should be alphabetical strings (no spaces, symbols and numbers).');
  }

  $encryptedText = strtolower($encryptedText);
  $key           = strtolower($key);
  $plainText     = [];
  $i             = 0;

  foreach (str_split($encryptedText) as $character) {
   array_push($plainText, $this->decryptCharacter($character, $key[$i]));
   $i++;
   $i = $i % strlen($key);
  }
  return implode('', $plainText);

 }

 private function normalizeAscii($character) {
  return ord($character) - ord('a');
 }

 private function returnAscii($normalizedAscii) {
  return chr($normalizedAscii + ord('a'));
 }

 private function encryptCharacter($plainCharacter, $keyCharacter) {
  return $this->returnAscii(($this->normalizeAscii($plainCharacter) + $this->normalizeAscii($keyCharacter)) % 26);
 }

 private function decryptCharacter($encryptedCharacter, $keyCharacter) {
  return  $this->returnAscii(($this->normalizeAscii($encryptedCharacter) - $this->normalizeAscii($keyCharacter) + 26) % 26);
 }
}

/* Example usage: 

$cipher    = new VigenereCipher();

$secret    = 'mysecret';
  
$encrypted = $cipher->encrypt('loremipsumdolor', $secret);

$plain     = $cipher->decrypt($encrypted, $secret);

echo $plain;

*/

File: code/cryptography/src/vigenere_cipher/vigenere_cipher.java
/**
 * Modified code from: https://rosettacode.org/wiki/Vigen%C3%A8re_cipher#Java
 * Part of Cosmos by Open Genus Foundation
 */

import java.util.Scanner;

public class vigenere_cipher{
    public static void main(String[] args) {
        System.out.println("Enter key: ");
        Scanner scanner = new Scanner(System.in);
        String key = scanner.nextLine();
        
        System.out.println("Enter text to encrypt: ");
        String text = scanner.nextLine();

        String enc = encrypt(text, key);
        System.out.println(enc);
        System.out.println(decrypt(enc, key));
    }
 
    static String encrypt(String text, final String key) {
        String res = "";
        for (int i = 0, j = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            if (Character.isUpperCase(c)){
              res += (char)((c + key.charAt(j) - 2 * 'A') % 26 + 'A');
              j = ++j % key.length();
            }
            if (Character.isLowerCase(c)){
              res += (char)((c + Character.toLowerCase(key.charAt(j)) - 2 * 'a') % 26 + 'a');
              j = ++j % key.length();
            }
        }
        return res;
    }
 
      static String decrypt(String text, final String key) {
          String res = "";
          for (int i = 0, j = 0; i < text.length(); i++) {
              char c = text.charAt(i);
              if (Character.isUpperCase(c)) {
                res += (char)((c - key.charAt(j) + 26) % 26 + 'A');
                j = ++j % key.length();
              }
              if (Character.isLowerCase(c)) {
                res += (char)((c - Character.toLowerCase(key.charAt(j)) + 26) % 26 + 'a');
                j = ++j % key.length();
              }
          }
          return res;
      }
}